### PR TITLE
Allow specifying topic type for SNS module.

### PR DESCRIPTION
**Pull request by stefanhorning**
##### SUMMARY
Adding a topic_type param to SNS module (simillar to SQS).

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
sns_topic module

##### ADDITIONAL INFORMATION
Simillar to SQS queues, SNS topics also allow specifying a type, that can either be 'Standard' or 'FIFO'. Furthermore if you have a FIFO SQS queue and want to subscribe that to an SNS topic, that one has to be FIFO too. Hence adding this flag is important and is actually just another option for the creat_topic action in Boto, see https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sns.html#SNS.Client.create_topic

### CODE REVIEW

File: plugins/modules/sns_topic.py [hunk 1]
```diff
@@ -287,9 +295,17 @@ def _topic_arn_lookup(self):
                 return topic
 
     def _create_topic(self):
+        attributes = { 'FifoTopic': 'false' }
```
Comment (stefanhorning) [inline plugins/modules/sns_topic.py:298]: A bunch of other attributes could go in here, but are alrady taken care of in the update methods further down that also to validity checking etc., so I left it at that for a create.

File: plugins/modules/sns_topic.py [hunk 2]
```diff
@@ -287,9 +295,17 @@ def _topic_arn_lookup(self):
...
+        tags = []
```
Comment (stefanhorning) [inline plugins/modules/sns_topic.py:299]: just to showcase that the create method can also take some tags. Feature could be added later

File: plugins/modules/sns_topic.py [hunk 3]
```diff
@@ -24,6 +24,12 @@
       - The name or ARN of the SNS topic to manage.
     required: true
     type: str
+  topic_type:
+    description:
+      - The type of topic that should be created. Either Standard for FIFO (first-in, first-out)
+    choices: ['standard', 'fifo']
+    default: 'standard'
+    type: str
```
Comment (alinabuzachis) [inline plugins/modules/sns_topic.py:32]: Could you please specify `version_added: 2.0.0`?

**Review by alinabuzachis — COMMENTED** [review]
@stefanhorning Thank you for working on this. Could you add please some integration tests? In addition, a changelog fragment is also required https://docs.ansible.com/ansible/latest/dev_guide/testing/sanity/changelog.html.

File: plugins/modules/sns_topic.py (hunk 3 above)
Comment (stefanhorning) [reply plugins/modules/sns_topic.py:32]: Done

**Comment by stefanhorning** [comment]
@alinabuzachis done now. Let's see if all tests pass.

**Comment by stefanhorning** [comment]
Build fails in this line https://github.com/ansible/ansible-zuul-jobs/blob/master/roles/ansible-test/tasks/split_targets.yaml#L23, not sure how that's related to my change (if at all).

**Comment by stefanhorning** [comment]
it was probably more likely that this was the change that broke CI (20 hours ago) https://github.com/ansible/ansible-zuul-jobs/commit/f8fa3e91e00b6ffb7a715dad370d7cc9cce945f2#diff-1e727d74735315c5405dbbc015402f13da5e0c3d8d7a08cae148cc7dac1c4f56

**Comment by tremble** [comment]
Looks like the failure occurs when there's not matching tests. a `default([])` is needed

**Comment by stefanhorning** [comment]
Also just noticed that the tests for sns_topic are disabled anyway with alias `unsupported`. So I guess we need to enable the test first before it will be run at all. Hence the test failures from CI are unrelated to this PR.

**Comment by stefanhorning** [comment]
There was a PR already to attempt enabling tests https://github.com/ansible-collections/community.aws/pull/519/files

I will just do that in here now to see how it goes @tremble

**Comment by tremble** [comment]
```
    "msg": "Unsupported parameters for (sns_topic) module: type. Supported parameters include: topic_type, aws_ca_bundle, aws_access_key (access_key, ec2_access_key), aws_secret_key (ec2_secret_key, secret_key), display_name, name, ec2_url (aws_endpoint_url, endpoint_url), profile (aws_profile), region (aws_region, ec2_region), security_token (access_token, aws_security_token), validate_certs, aws_config, state, delivery_policy, purge_subscriptions, subscriptions, policy, debug_botocore_endpoint_logs."
```

**Comment by stefanhorning** [comment]
should read my own docs :) 
Fixed now

File: tests/integration/targets/sns_topic/aliases [hunk 4]
```diff
@@ -1,4 +1,2 @@
-# reason: missing-policy
-unsupported
-
+shippable/aws/group1
```
Comment (tremble) [inline tests/integration/targets/sns_topic/aliases:1]:
```suggestion
```

**Review by tremble — APPROVED** [review]
I'm sorry about the delay here.  The main change looks good.

2 minor things:

- defaults/main.yml in the test has a merge conflict (side effect of the zuul migration)
- shippable/aws/group1 is no longer needed in the aliases file, just remove the unsupported entry

**Comment by stefanhorning** [comment]
@tremble merged in latest main now and resolved conflicts, or should I rather rebase?

**Comment by tremble** [comment]
Please rebase if possible.  With the migration to Zuul commits are now merged as-is rather than squashed.

**Comment by stefanhorning** [comment]
Yes, the bot already answered my question ;)

Rebased now

File: changelogs/fragments/sns_topic_type.yml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - sns_topic - Added ``topic_type`` parameter to select type of SNS topic (either FIFO or Standard)
```
Comment (tremble) [inline changelogs/fragments/sns_topic_type.yml:2]:
```suggestion
  - sns_topic - Added ``topic_type`` parameter to select type of SNS topic (either FIFO or Standard) (https://github.com/ansible-collections/community.aws/pull/599).
```

**Review by tremble — CHANGES_REQUESTED** [review]
apologies, one minor nit and we're good to merge.

**Comment by stefanhorning** [comment]
done

**Review by tremble — APPROVED** [review]

**Comment by tremble** [comment]
@stefanhorning Thanks for your contribution and patience.

**Pull request merged**